client/rest-high-level/src/test/java/org/elasticsearch/client/DocumentationIT.java

```
@@ -0,0 +1,84 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.delete.DeleteRequest;
+import org.elasticsearch.action.delete.DeleteResponse;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.VersionType;
+
+import java.io.IOException;
+
+/**
+ * This class is used to generate the Java API documentation
+ */
+public class DocumentationIT extends ESRestHighLevelClientTestCase {
+
+    /**
+     * This test documents docs/java-rest/high-level/document/delete.asciidoc
+     */
+    public void testDelete() throws IOException {
+        RestHighLevelClient client = highLevelClient();
+
+// tag::delete-request[]
+DeleteRequest request = new DeleteRequest(
+    "index",    // <1>
+    "type",     // <2>
+    "id");      // <3>
+// end::delete-request[]
+
+// tag::delete-request-props[]
+request.timeout(TimeValue.timeValueSeconds(1));                     // <1>
+request.timeout("1s");                                              // <2>
+request.setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);    // <3>
+request.setRefreshPolicy("wait_for");                               // <4>
+request.version(2);                                                 // <5>
+request.versionType(VersionType.EXTERNAL);                          // <6>
+// end::delete-request-props[]
+
+// tag::delete-execute[]
+DeleteResponse response = client.delete(request);
+// end::delete-execute[]
+
+// tag::delete-notfound[]
+if (response.getResult().equals(DocWriteResponse.Result.NOT_FOUND)) {
+    // <1>
```

Maybe throw an exception from here just so the docs look more realistic?